packages/twenty-server/src/engine/api/rest/core/query-builder/core-query-builder.factory.ts

```diff
@@ -119,15 +134,69 @@ export class CoreQueryBuilderFactory {
     };
   }
 
-  async createMany(request: Request): Promise<Query> {
+  async createMany(request: Request): Promise<QueryVariables> {
     const { object: parsedObject } = parseCoreBatchPath(request);
     const objectMetadata = await this.getObjectMetadata(request, parsedObject);
     const depth = computeDepth(request);
 
-    return {
-      query: this.createManyQueryFactory.create(objectMetadata, depth),
-      variables: this.createVariablesFactory.create(request),
+    const context: AuthContext = {
+      user: request.user,
+      apiKey: request.apiKey,
+      workspaceMemberId: request.workspaceMemberId,
+      workspace: request.workspace as Workspace,
     };
+
+    const objectMetadataMap =
+      await this.workspaceStorageCacheService.getObjectMetadataMap(
+        context.workspace.id,
+        request.workspaceMetadataVersion as number,
+      );
+
+    if (!objectMetadataMap) {
+      await this.workspaceMetadataCacheService.recomputeMetadataCache({
+        workspaceId: context.workspace.id,
+      });
+      throw new Error('Object metadata collection not found');
+    }
+
+    const objectMetadataMapItem =
+      objectMetadataMap[objectMetadata.objectMetadataItem.nameSingular];
+
+    const dataSource =
+      await this.twentyORMGlobalManager.getDataSourceForWorkspace(
+        context.workspace.id,
+      );
+
+    const repository = dataSource.getRepository(
+      objectMetadataMapItem.nameSingular,
+    );
+
+    const objectRecords: InsertResult = !request.body.upsert
+      ? await repository.insert(request.body)
+      : await repository.upsert(request.body, {
+          conflictPaths: ['id'],
+          skipUpdateIfNoValuesChanged: true,
+        });
+
+    const queryBuilder = repository.createQueryBuilder(
+      objectMetadataMapItem.nameSingular,
+    );
+
+    const nonFormattedUpsertedRecords = await queryBuilder
+      .where({
+        id: In(objectRecords.generatedMaps.map((record) => record.id)),
+      })
+      .take(QUERY_MAX_RECORDS)
+      .getMany();
+
+    const upsertedRecords = formatResult(
+      nonFormattedUpsertedRecords,
+      objectMetadataMapItem,
+      objectMetadataMap,
+    );
```

Comment:
This code shared lot of code with `packages/twenty-server/src/engine/api/graphql/graphql-query-runner/resolvers/graphql-query-create-many-resolver.service.ts`
We must factorize that shared code